airbyte-integrations/connectors/source-twilio/source_twilio/auth.py

```diff
@@ -0,0 +1,35 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import base64
+from typing import Tuple
+
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+class HttpBasicAuthenticator(TokenAuthenticator):
+    def __init__(self, auth: Tuple[str, str], auth_method: str = "Basic", **kwargs):
+        auth_string = f"{auth[0]}:{auth[1]}".encode("utf8")
+        b64_encoded = base64.b64encode(auth_string).decode("utf8")
+        super().__init__(token=b64_encoded, auth_method=auth_method, **kwargs)
```

Comment:
Well, I can, but then I need to put base64 encoding out of auth class. Do you think it's a better approach or have some other ideas?

Reply:
I think, config transformation can be implemented in Source.

Reply:
Do we have some similar examples in the project? Because if we put it in source, then we'll need to duplicate code, at least in `check_connection` and `streams` methods in `SourceTwilio` class.

Reply:
You can create Source method for this as I did in Jira Source.